testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js

```diff
@@ -602,6 +602,25 @@ QUnit.test("valueChanged event fires on value apply", function(assert) {
     assert.ok(valueChangedSpy.calledOnce, "valueChanged event called once");
 });
 
+QUnit.testInActiveWindow("caret position is not changed when the focus out event has occurred", function(assert) {
+    var _caret = this.instance._caret,
+        caretIsUpdatedOnFocusOut,
+        $input = $(this.instance.element()).find(".dx-texteditor-input");
+
+    this.instance._caret = (function(newCaret) {
+        var result = _caret.apply(this.instance, newCaret);
+        if(!$input.is(":focus") && newCaret) {
+            caretIsUpdatedOnFocusOut = true;
+        }
+        return result;
+    }).bind(this);
+    this.keyboard.type(".").blur();
+
+    this.instance._caret = _caret;
```

Comment:
It is better to use try-finally construction here. It will prevent next tests from a crash if this test crash.

Reply:
The new instance of NumberBox is created for an each test